apache2/apache2_util.c

```diff
@@ -187,6 +192,192 @@ char *get_env_var(request_rec *r, char *name) {
     return result;
 }
 
+#ifdef WAF_JSON_LOGGING_ENABLE
+/**
+ * Retrieve waf log field.
+ */
+void get_field_value(const char* from, const char* to, const char* text, char* output) {
+    char* first = strstr(text, from);
+    int first_index = first - text;
+
+    if (first != NULL ) {
+        if ((first_index > 0 && (first-1)[0] != "\\") || (first_index == 0)) {
+            first += strlen(from);
+        }
+        else {
+            first = NULL;
+        }
+    }
+    else {
+        return;
+    }
+
+    char* last = strstr(first, to);
+    int last_index = last- first;
+    if (last != NULL ) {
+        if ((last_index > 0 && (last-1)[0] != "\\") || (last_index == 0)) {
+        }
+        else {
+            last = NULL;
+        }
+    }
+
+    if (first != NULL && last != NULL) {
+        strncpy(output, first, last - first);
+    }
+}
+
+/**
+ * get ip and port number.
+ */
+void get_ip_port(const char* ip_port, char* waf_ip, char* waf_port) {
```

Comment:
@azhao155 could you make this `static`?

Reply:
will change to the static for visibility.